### ISSUE TITLE
The summary for converted ActivityPub objects must be treated as html according to the ActivityStreams specification

### ISSUE THREAD

**Issue by Menrath** [mastodon/mastodon#28455]
### Steps to reproduce the problem

1. Retrieve a non-note object like an `Event` from with a HTML summary.
2. The resulting post will not be rendered/converted as HTML but as plain text.

### Expected behaviour

The summary should be treated as HTML by default. 

### Actual behaviour

The summary is treated as plain text. 

The following Screenshot shows what a event received from Gancio then looks like.
![Screenshot_from_2023-12-20_01-20-36](https://github.com/mastodon/mastodon/assets/99024746/d9472d8a-2e0c-4648-b1e5-574428110565)

### Detailed description

The feature introduced in https://github.com/mastodon/mastodon/pull/9823 is basically a great one, without Mastodon implementing custom renderings for ActivityPub objects like `Event` it has basic support for them. The actual convertion likely takes place here: https://github.com/mastodon/mastodon/blob/2463b53363a621c1e18223bda5c47a663707a22c/app/lib/activitypub/activity/create.rb#L371-L373.

The [ActivityStreams vocabulary specification](https://www.w3.org/TR/activitystreams-vocabulary/#dfn-summary) states that 

> **Summary**: A natural language summarization of the object encoded as HTML. 

This will cause interoperability issues with future Gancio, Mobilizon and WordPress ActivityPub releases.

Altough the ActivityPub spec does not mention it, one might also think of accepting custom `MediaType` for the summary, to handle both plain/text and HTML summaries.

See https://framagit.org/les/gancio/-/issues/321#note_2038846

### Mastodon instance

All

### Mastodon version

Since https://github.com/mastodon/mastodon/pull/9823

**Comment by KDederichs** [comment]
Correct me if I'm wrong here but:

Rendering any HTML form an unsafe source would open up security risks (XSS and such).

While it's in the spec, I doubt it's a good idea to just render any HTML that comes in as is.

IF one would implement this you'd have to very carefully try and sanitize it of any malicious stuff (like scripts) imo.

**Comment by Menrath** [comment]
> Correct me if I'm wrong here but:
> 
> Rendering any HTML form an unsafe source would open up security risks (XSS and such).
> 
> While it's in the spec, I doubt it's a good idea to just render any HTML that comes in as is.
> 
> IF one would implement this you'd have to very carefully try and sanitize it of any malicious stuff (like scripts) imo.

The default content of the ActivityPub object is already treated as HTML within Mastodon. Not sure though at which stage the sanitization takes place.

**Comment by trwnh** [comment]
> The [summary](https://www.w3.org/TR/activitystreams-vocabulary/#dfn-summary) property should be used as a fallback text representation, possibly automatically generated by the publisher. If there is no name property, the summary property should not include markup, and should be short enough to be used as a reasonable text representation of the object.

from as2-core section 4.1.1

**Comment by Menrath** [comment]
> > The [summary](https://www.w3.org/TR/activitystreams-vocabulary/#dfn-summary) property should be used as a fallback text representation, possibly automatically generated by the publisher. If there is no name property, the summary property should not include markup, and should be short enough to be used as a reasonable text representation of the object.
> 
> from as2-core section 4.1.1

And have you come to any different conclusions than I have?

**Comment by ClearlyClaire** [comment]
Do you have a link to an example activity that triggers the behavior shown in the screenshot?

**Comment by Menrath** [comment]
> Do you have a link to an example activity that triggers the behavior shown in the screenshot?

@lesion Made that specific test together with the screenshot, maybe he can provide the JsonLD of the Activity it shows.

**Comment by lesion** [comment]
> @lesion Made that specific test together with the screenshot, maybe he can provide the JsonLD of the Activity it shows.

This was a test to check if Mastodon will render the summary as html, something like:

```  json
{
         "actor" : "https://test.cisti.org/federation/u/gancio",
         "id" : "https://test.cisti.org/federation/m/3#create",
         "object" : {
            "attachment" : [
               {
                  "focalPoint" : [
                     0,
                     0
                  ],
                  "mediaType" : "image/jpeg",
                  "name" : "AP test",
                  "type" : "Document",
                  "url" : "https://test.cisti.org/media/e10f1e2e1724a6104b7fed8b46db1f86.jpg"
               }
            ],
            "attributedTo" : "https://test.cisti.org/federation/u/gancio",
            "cc" : [
               "https://test.cisti.org/federation/u/gancio/followers"
            ],
            "content" : "<p>test <strong>description</strong></p>",
            "id" : "https://test.cisti.org/federation/m/3",
            "location" : {
               "address" : "Piazza Castello 1, Torino",
               "name" : "Piazza Castello",
               "type" : "Place"
            },
            "name" : "AP test",
            "published" : "2024-01-08T11:08:13.261Z",
            "startTime" : "2024-01-31T15:15:00.000+01:00",
            "summary" : "online, <u>Wednesday, 31 January</u> (15:15)",
            "tag" : [
               {
                  "href" : "https://test.cisti.org/tag/test",
                  "name" : "#test",
                  "type" : "Hashtag"
               }
            ],
            "to" : [
               "https://www.w3.org/ns/activitystreams#Public"
            ],
            "type" : "Event",
            "url" : "https://test.cisti.org/event/ap-test"
         },
         "published" : "2024-01-08T17:23:08.378+01:00",
         "to" : "https://www.w3.org/ns/activitystreams#Public",
         "type" : "Create"
      },
```

**Comment by pfefferle** [comment]
I currently have the same issue with `Articles`.

It seems that Mastodon also encodes the HTML in the summary for `Articles`. I have no emotions about if Mastodon should support HTML in the summary or not, but I would vote for replacing the HTML instead of encoding it!
